thorlcr/activities/indexread/thindexread.cpp

```diff
@@ -207,7 +216,7 @@ class CIndexReadBase : public CMasterActivity
             if (!isFileKey(index))
                 throw MakeActivityException(this, 0, "Attempting to read flat file as an index: %s", helperFileName.get());
 
-            localKey = index->queryAttributes().getPropBool("@local");
+            localKey = index->queryAttributes().getPropBool("@local") && !index->queryAttributes().hasProp("@partitionFieldMask");;
```

Comment:
.. also (not new) think that this : 
```
                if (container.queryLocalOrGrouped())
                    nofilter = true;
```
is setting nofilter=true, if local _activity_ (not localKey), I'm not sure it's right in general, but for a partitionKey, it will prevent the performPartLookup filtering, i.e. it will send all parts, so I think it should be changed to:
```
                if (container.queryLocalOrGrouped() && !partitionKey)
                    nofilter = true;
```

Reply:
I don't really understand why the restriction:

                throw MakeActivityException(this, 0, "Index Read cannot be LOCAL unless supplied index is local");

makes any sense, to be honest. Any key could be treated as a local key just by ignoring the information in the TLK...

Reply:
"But even if 1 part, it could be filtered out..."
 
It's going to do a filter on the slave anyway - no point also doing one on the master.

Reply:
"I'm not sure it's right in general, but for a partitionKey, it will prevent the performPartLookup filtering, i.e. it will send all parts"

If it's wrong for partition keys it's wrong for keys with a full TLK too - they also should be filtering (and in general will do so more effectively than partition keys)

Reply:
agreed, separate issue then, can investigate via separate JIRA.

Reply:
>It's going to do a filter on the slave anyway - no point also doing one on the master.

well, it's going out of it's way to not filter it, and hence serializing unnecessary info. to slaves.
But agree not a big deal.

Reply:
>makes any sense, to be honest. Any key could be treated as a local key just by ignoring the information in the TLK...

Not really sure, but not related to this PR.